activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -78,7 +78,7 @@ def initialize(constructor = {})
     def default(*args)
       arg_key = args.first
 
-      if include?(key = convert_key(arg_key))
+      if include?(key = convert_key(arg_key)) && args.size > 0
         self[key]
```

Comment:
Yes it does, but otherwise `HashWithIndifferentAccess#dig` doesn't work. #23414 fixes #18909.